### DISCUSSION TITLE
control robot joint/link goal position with close precision

### DISCUSSION THREAD

**Discussion by shivanimall** [bulletphysics/bullet3#3827] [General]
Hello @erwincoumans 

Here is some background-

I have a robot with only a right hand with one finger tip. This finger tip say `finger_right_tip` is defined as a link and controlled by a joint say `guide_joint_finger_right`.  The `finger_right_tip` is loaded with a sensor object. 
However,  I wasn't able to control the actual location of the `finger_right_tip`(which is a link) with close accuracy. 

My end goal is to make the  sensor object loaded on the `finger_right_tip` to reach a desired base position and orientation with close accuracy( upto one decimal).  Is it possible for me to send you small snippet for you to take a look?

**Comment by shivanimall** [comment]
Hello Erwin, following up on this- to give you some more background, linking detailed issue:
facebookresearch/tacto#18

The understanding is that pybullet IK solver maybe not accurate enough for my requirement to reach the desired goal/base position with precision and as previously discussed [here](https://github.com/bulletphysics/bullet3/discussions/3804) specifying collision points via `getContactPoints` isn't supported.

**Comment by erwincoumans** [comment]
The PyBullet inverse kinematics is very precise (sub millimeter should be easily doable), not likely the issue. If needed, you can increase the IK solver iterations (unlikely the cause). You can use resetState, and getLinkState (enable compute forward kinematics) to compute the error between actual link pos and target link pos.

You may give the simulation some time to reach the target, when using position control (setJointMotorControl2)

 See also https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/examples/inverse_kinematics.py
another example:
https://github.com/bulletphysics/bullet3/tree/master/examples/pybullet/gym/pybullet_robots/panda

Unfortunately, there is no paid support, I'm busy with other projects, so you'll have to figure it out by yourself (or someone else not me).
Good luck!
